File: plan-service/src/main/java/scdy/planservice/common/advice/ControllerAdvice.java
package scdy.planservice.common.advice;


import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.ExceptionHandler;
import org.springframework.web.bind.annotation.RestControllerAdvice;
import scdy.planservice.common.exceptions.*;

@RestControllerAdvice
public class ControllerAdvice {
    @ExceptionHandler(NotFoundException.class) // 404 Not Found
    public ResponseEntity<ErrorResponse> handleNotFoundException(NotFoundException e) {
        return ResponseEntity.status(HttpStatus.NOT_FOUND).body(new ErrorResponse(e.getMessage()));
    }

    @ExceptionHandler(BadRequestException.class) // 400 Bad Request
    public ResponseEntity<ErrorResponse> handleBadRequestException(BadRequestException e) {
        return ResponseEntity.badRequest().body(new ErrorResponse(e.getMessage()));
    }
    @ExceptionHandler(ForbiddenException.class) // 403 Forbidden
    public ResponseEntity<ErrorResponse> handleForbiddenException(ForbiddenException e) {
        return ResponseEntity.status(HttpStatus.FORBIDDEN).body(new ErrorResponse(e.getMessage()));
    }


    @ExceptionHandler(UnauthorizedException.class) // 401 Unauthorized
    public ResponseEntity<ErrorResponse> handleUnauthorizedException(UnauthorizedException e) {
        return ResponseEntity.status(HttpStatus.UNAUTHORIZED).body(new ErrorResponse(e.getMessage()));
    }

    @ExceptionHandler(InternalServerException.class) //500 Internal Server Error
    public ResponseEntity<ErrorResponse> handleInternalServerException(InternalServerException e) {
        return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(new ErrorResponse(e.getMessage()));
    }

}

File: board-service/src/main/kotlin/scdy/boardservice/common/advice/ControllerAdvice.kt
package scdy.boardservice.common.advice

import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.ExceptionHandler
import org.springframework.web.bind.annotation.RestControllerAdvice
import scdy.boardservice.common.exceptions.*

@ExceptionHandler(BadRequestException::class)
private // 400 Bad Request
fun handleBadRequestException(e: BadRequestException): ResponseEntity<ErrorResponse> {
    return ResponseEntity.badRequest().body(ErrorResponse(e.message?: "Bad Request"))
}

@RestControllerAdvice
class ControllerAdvice {

    @ExceptionHandler(NotFoundException::class) // 404 Not Found
    fun handleNotFoundException(e: NotFoundException): ResponseEntity<ErrorResponse> {
        return ResponseEntity.status(HttpStatus.NOT_FOUND).body(ErrorResponse(e.message?: "Bad Request"))
    }

    @ExceptionHandler(BadRequestException::class)
    private // 400 Bad Request
    fun handleBadRequestException(e: BadRequestException): ResponseEntity<ErrorResponse> {
        return ResponseEntity.badRequest().body(ErrorResponse(e.message?: "Bad Request"))
    }

    @ExceptionHandler(ForbiddenException::class) // 403 Forbidden
    fun handleForbiddenException(e: ForbiddenException): ResponseEntity<ErrorResponse> {
        return ResponseEntity.status(HttpStatus.FORBIDDEN).body(ErrorResponse(e.message?: "Bad Request"))
    }

    @ExceptionHandler(UnauthorizedException::class) // 401 Unauthorized
    fun handleUnauthorizedException(e: UnauthorizedException): ResponseEntity<ErrorResponse> {
        return ResponseEntity.status(HttpStatus.UNAUTHORIZED).body(ErrorResponse(e.message?: "Bad Request"))
    }

    @ExceptionHandler(InternalServerException::class) // 500 Internal Server Error
    fun handleInternalServerException(e: InternalServerException): ResponseEntity<ErrorResponse> {
        return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(ErrorResponse(e.message?: "Bad Request"))
    }
}